lib/stdlib.ngs

```diff
@@ -2105,6 +2105,22 @@ section "Range" {
 	TEST 1 =~ 0..5
 	TEST 0 =~ 0..5
 	TEST 5 !~ 0..5
+
+	doc Ensure that n is in the range. 
+	doc n in range returns n
+	doc %EX - ensure(1, 3..10) #3
+	F ensure(n:Int, r:NumRange){
+
+    	if n in r {
+        	n
+    	} else if n > r.end {
+        	r.end
```

Comment:
* This is only correct if `r.include_end`. Otherwise, it's just outside the range. Should probably be `r.end - r.step` when `not(r.include_end)`. Same for start of the range (with appropriate adaptation).
* Please add `TEST ...` lines with tests

Reply:
what you mean by r.end - r.step ???

Reply:
The exact expression `r.end - r.step` to do determine last number in the range (when `r.include_end` is `false`)